src/writer/emitter.rs

```diff
@@ -42,18 +42,31 @@ impl From<io::Error> for EmitterError {
 
 impl fmt::Display for EmitterError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use std::error::Error;
+
         try!(write!(f, "emitter error: "));
         match *self {
             EmitterError::Io(ref e) =>
                 write!(f, "I/O error: {}", e),
+            ref other =>
+                write!(f, "{}", other.description()),
+        }
+    }
+}
+
+impl ::std::error::Error for EmitterError {
```

Comment:
Please put `use ::std::error::Error` to the top of the file and use just `Error` here.